packages/jest-mock/src/__tests__/index.test.ts

```diff
@@ -238,20 +238,13 @@ describe('moduleMocker', () => {
       }
 
       const ClassFooMock = moduleMocker.generateFromMetadata(
-        moduleMocker.getMetadata(ClassFoo),
+        moduleMocker.getMetadata(ClassFoo)!,
       );
       const foo = new ClassFooMock();
-
-      const instanceFoo = new ClassFoo();
-      const instanceFooMock = moduleMocker.generateFromMetadata(
-        moduleMocker.getMetadata(instanceFoo),
-      );
-
       expect(typeof foo.foo).toBe('function');
-      expect(typeof instanceFooMock.foo).toBe('function');
-      expect(instanceFooMock.foo.mock).toBeDefined();
+      expect(foo.foo.mock).toBeDefined();
 
-      expect(instanceFooMock.toString.mock).toBeDefined();
+      expect(foo.toString.mock).toBeDefined();
```

Comment:
modified this test a little to use mocked class , initially it is passing the instance created from original class to get meta data (may be due to this issue #14458)

Reply:
type tests are failing :(  related to this PR , don't know why they pass in my local

Reply:
@mrazauskas  would you like to review this PR?

Reply:
> type tests are failing :( related to this PR , don't know why they pass in my local

I have just left a https://github.com/jestjs/jest/pull/15179#discussion_r1673946100. Hope this helps.